dart/test/http_client/sentry_http_client_test.dart

```diff
@@ -183,10 +176,37 @@ void main() {
         expect(e.message, 'test');
       }
 
-      expect(mockHub.addBreadcrumbCalls.length, 0);
       expect(mockHub.captureExceptionCalls.length, 0);
     });
 
+    test('breadcrumb gets added when an exception gets thrown', () async {
+      final url = Uri.parse('https://example.com');
+
+      final mockHub = MockHub();
+
+      final mockClient = MockClient((request) async {
+        expect(request.url, url);
+        throw Exception('foo bar');
+      });
+
+      final client = SentryHttpClient(client: mockClient, hub: mockHub);
+
+      try {
+        await client.get(Uri.parse('https://example.com'));
+        fail('Method did not throw');
+      } on Exception catch (_) {}
```

Comment:
not sure if all but most of this common code across tests could be improved with the fixture pattern, lets always try to keep it clean, so when touching test classes that don't have it, we try to migrate to not duplicate even more things.

Reply:
Yup, already on it. I knew you would say that 👍